storybookjs/storybook
# Core: Allow `.mjs` extension for CSF stories

Closes #23078 

## What I did

- `lib/core-common`
  - Added "mjs" to `normalizeStoriesEntry` default files glob pattern
  - Updated relevant inline snapshots
- `lib/core-server`
  - Updated the regex test for the `csfIndexer` to allow ESM `.mjs` type story files.
  - Added mock data for existing `StoryIndexGenerator` & `stories-json` tests
  - Updated relevant inline snapshots & expected test values
- `lib/cli`
  - Added "mjs" to the `configureMain` extensions
  - Updated relevant inline snapshots
- `ui`
  - Added "mjs" to `blocksOnlyStories` glob patterns
- Updated stories file extensions lists in docs

## How to test

1. Run storybook with `yarn start`
2. Run `yarn test`
3. Ensure the following tests pass:
  - `lib/core-common/src/utils/__tests__/normalize-stories.test.ts`
  - `lib/core-server/src/utils/StoryIndexGenerator.test.ts`
  - `lib/core-server/src/utils/stories-json.test.ts`
  - `lib/cli/src/automigrate/fixes/bare-mdx-stories-glob.test.ts`
  - `lib/cli/src/generators/configure.test.ts`

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
